pkg/apis/keda/v1alpha1/triggerauthentication.go

```diff
@@ -0,0 +1,64 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TriggerAuthentication defines how a trigger can authenticate
+type TriggerAuthentication struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec TriggerAuthenticationSpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TriggerAuthenticationList is a list of TriggerAuthentication resources
+type TriggerAuthenticationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []TriggerAuthentication `json:"items"`
+}
+
+// TriggerAuthenticationSpec defines the various ways to authenticate
+type TriggerAuthenticationSpec struct {
+	// +optional
+	PodIdentity bool `json:"podIdentity"`
+
+	// +optional
+	SecretTargetRef []AuthSecretTargetRef `json:"secretTargetRef"`
+
+	// +optional
+	Env []AuthEnvironment `json:"env"`
+}
+
+// AuthPodIdentity allows users to select the platform native identity
+// mechanism
+type AuthPodIdentity struct {
+	Provider string `json:"provider"`
```

Comment:
Is there a way in Go that allows us to mark it as an enum? Otherwise all scalers will have to provide the same logic to determine the supported providers which means a lot of duplication, no?

Reply:
@tomkerkhove there are no enums in go. But I've done the next closest thing.

Reply:
What?! Lol! :D